src/pages/SupplierRequisitionPage.js

```diff
@@ -1,666 +1,334 @@
-          info: period && `${period.name} -- ${period.toString()}`,
-          shouldHide: !program,
-        },
-        {
-          title: `${pageInfoStrings.supplier}:`,
-          info: requisition.otherStoreName ? requisition.otherStoreName.name : '',
-        },
-        {
-          title: `${pageInfoStrings.months_stock_required}:`,
-          info: requisition.monthsToSupply,
-          onPress: !program && this.openMonthsSelector,
-          editableType: !program && 'selectable',
-        },
-        {
-          title: `${pageInfoStrings.comment}:`,
-          info: requisition.comment,
-          onPress: this.openCommentEditor,
-          editableType: 'text',
-        },
-      ],
-    ];
-
-    return <PageInfo columns={infoColumns} isEditingDisabled={requisition.isFinalised} />;
-  };
-
-  renderCell = (key, requisitionItem) => {
-    const { requisition } = this.props;
-    switch (key) {
-      default:
-        return requisitionItem[key];
-      case 'monthlyUsage':
-      case 'suggestedQuantity':
-        return Math.round(requisitionItem[key]);
+/**
+ * Renders a mSupply mobile page with a supplier requisition loaded for editing
+ *
+ * State:
+ * Uses a reducer to manage state with `backingData` being a realm results
+ * of items to display. `data` is a plain JS array of realm objects. data is
+ * hydrated from the `backingData` for displaying in the interface.
+ * i.e: When filtering, data is populated from filtered items of `backingData`.
+ *
+ * dataState is a simple map of objects corresponding to a row being displayed,
+ * holding the state of a given row. Each object has the shape :
+ * { isSelected, isDisabled },
+ *
+ * @prop {Object} requisition The realm transaction object for this invoice.
+ * @prop {Func}   runWithLoadingIndicator Callback for displaying a fullscreen spinner.
+ * @prop {String} routeName The current route name for the top of the navigation stack.
+ */
+export const SupplierRequisitionPage = ({ requisition, runWithLoadingIndicator, routeName }) => {
+  const [state, dispatch, instantDebouncedDispatch, debouncedDispatch] = usePageReducer(routeName, {
+    pageObject: requisition,
+    backingData: requisition.items,
+    data: requisition.items.sorted('item.name').slice(),
+    keyExtractor: recordKeyExtractor,
+    dataState: new Map(),
+    currentFocusedRowKey: null,
+    searchTerm: '',
+    filterDataKeys: ['item.name'],
+    sortBy: 'itemName',
+    isAscending: true,
+    modalKey: '',
+    hasSelection: false,
+    showAllStock: !(routeName === 'programSupplierRequisition'),
+  });
+
+  const { ITEM_SELECT, REQUISITION_COMMENT_EDIT, MONTHS_SELECT, VIEW_REGIMEN_DATA } = MODAL_KEYS;
+  const {
+    data,
+    dataState,
+    sortBy,
+    isAscending,
+    columns,
+    modalKey,
+    pageInfo,
+    pageObject,
+    hasSelection,
+    showAllStock,
+    keyExtractor,
+  } = state;
+
+  const { isFinalised, comment, theirRef, program } = pageObject;
+
+  useEffect(() => {
```

Comment:
Should use state initializer function in order to not need this, can also be user by the reducer to reset state for i.e. after finalizing?